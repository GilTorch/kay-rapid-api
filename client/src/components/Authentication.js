import React from 'react';
import { Link } from 'react-router-dom';
import "../css/authentication.css";

class Authentication extends React.Component{
    constructor(props){
        super(props);
    }

    render(){
        return(
        <div className="card signin-card">
            <div className="signin-card__wrapper">
                <p className="signin-card__message">Konekte kounya pou ou ka genyen yon pi bon eksperyans.</p>
                <Link to="/authentication/sign-in-with-social-media">                                                    
                    <button className="auth-button success-button">KONEKTE</button>
                </Link>
            </div>
        </div>
        )
    }
}

export default Authentication;