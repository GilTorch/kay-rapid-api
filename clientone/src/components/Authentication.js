import React from 'react';
import { Link } from 'react-router-dom';

class Authentication extends React.Component{
   

    render(){
        return(
        <div className="card signin-card">
            <div className="signin-card__wrapper">
                <p>Konekte kounya pou ou ka genyen yon pi bon eksperyans.</p>
                <Link to="/authentication/sign-in-with-social-media">                                                    
                    <button className="authentication__button signin-card__button">KONEKTE</button>
                </Link>
            </div>
        </div>
        )
    }
}

export default Authentication;