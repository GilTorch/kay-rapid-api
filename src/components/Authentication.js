import React from 'react';
import SignIn from './SignIn';
import { BrowserRouter as Router,Route } from 'react-router-dom';
import SignInWithSocialMedia from './SignInWithSocialMedia';
import SignInWithoutSocialMedia from './SignInWithoutSocialMedia';
import AccountCreation from './AccountCreation';


class Authentication extends React.Component{

    constructor(props){
        super(props);
        this.closeCard=this.closeCard.bind(this);
        // $('.authentication__signin-with-socialmedia').hide();
        // $('.authentication__signin-without-socialmedia').hide();
        // $('.authenctication__account-creation').hide();
        this.state={
            userIsSignedIn:false,
        }
    }


    closeCard(){
        
    }

    render(){
       return(
        <div className="authentication">
        <Router>
        <React.Fragment>
          <Route exact={true} path="/authentication" component={SignIn}/>
          <Route exact={true} path="/authentication/sign-in-with-social-media" component={SignInWithSocialMedia}/>
          <Route exact={true} path="/authentication/sign-in-without-social-media" component={SignInWithoutSocialMedia}/>
          <Route exact={true} path="/authentication/account-creation" component={AccountCreation}/>
        </React.Fragment>
      </Router>
        </div>
        )
    }

}


export default Authentication;