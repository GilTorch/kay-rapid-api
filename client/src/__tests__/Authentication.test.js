import React from 'react';
import Authentication from '../components/Authentication';
import { shallow,mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter,Link } from 'react-router-dom';

describe("<Authentication/>",()=>{

    it ('should have a message that prompts users to connect',()=>{
        const wrapper=shallow(<Authentication/>)
        expect(wrapper.find('.signin-card__message')).toHaveLength(1)
    })

    it ('should have a connexion button',()=>{
        const wrapper=shallow(<Authentication/>)
        expect(wrapper.find('.success-button')).toHaveLength(1)
    })

    it ('should match the snapshot',()=>{
        const wrapper=renderer
        .create(
        <MemoryRouter initialEntries={['/profile']}>
            <Authentication/>
        </MemoryRouter>
        )
        .toJSON()
        expect(wrapper).toMatchSnapshot();
    })

    it ('should go to signin screen when clicked',()=>{
        const wrapper=mount(
                <Authentication/>
        )

        expect(wrapper.find(<Link to="/authentication/sign-in-with-social-media"/>)).toHaveLength(1)
        
    })
})