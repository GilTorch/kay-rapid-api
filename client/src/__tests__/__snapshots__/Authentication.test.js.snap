// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication/> should have match snapshot 1`] = `
<div
  className="card signin-card"
>
  <div
    className="signin-card__wrapper"
  >
    <p
      className="signin-card__message"
    >
      Konekte kounya pou ou ka genyen yon pi bon eksperyans.
    </p>
    <a
      href="/authentication/sign-in-with-social-media"
      onClick={[Function]}
    >
      <button
        className="auth-button success-button"
      >
        KONEKTE
      </button>
    </a>
  </div>
</div>
`;

exports[`<Authentication/> should match the snapshot 1`] = `
<div
  className="card signin-card"
>
  <div
    className="signin-card__wrapper"
  >
    <p
      className="signin-card__message"
    >
      Konekte kounya pou ou ka genyen yon pi bon eksperyans.
    </p>
    <a
      href="/authentication/sign-in-with-social-media"
      onClick={[Function]}
    >
      <button
        className="auth-button success-button"
      >
        KONEKTE
      </button>
    </a>
  </div>
</div>
`;
